File: go_version/trialDivision.go
package main

import (
  "time"
)

func trial_division(n int) ([]int, int) {
  defer TimeTrack(time.Now(), "Trial Division")

  var p_list []int
  var p = 2

  var it_count = 1
  for p*p <= n {
    for n % p == 0 {
      p_list = append(p_list, p)
      n /= p
    }
    p += 1

    it_count += 1
  }

  if ( n != 1 ) {
    p_list = append(p_list, n)
  }

  return p_list, it_count
}

func trial_division2(n int) ([]int, int) {
  defer TimeTrack(time.Now(), "Trial Division 2")
  var it_count = 1

  var p_list []int

  for n % 2 == 0 {
    p_list = append(p_list, 2)
    n /= 2

    it_count += 1
  }
  var p = 3

  for p * p <= n {
    if n % p == 0 {
      p_list = append(p_list, p)
      n /= p
    } else {
      p += 2
    }
    it_count += 1
  }
  if n != 1 {
    p_list = append(p_list, n)
  }

  // Only odd number is possible
  return p_list, it_count
}

File: trial_division.py
from decimal import *
import time


def trial_division(n):
	t = time.process_time_ns()

	p_list = []
	p = Decimal(2)

	it_count = 1
	while (p*p <= n):
		while (n%p == 0):
			p_list.append(p)
			n /= p

		p += 1
		
		it_count += 1
	
	if (n != 1):
		p_list.append(n)

	if (len(p_list) > 2):
	    print("AAAAAA")
	
	return int(p_list[0]), int(p_list[1]), it_count, time.process_time_ns()-t




def trial_division_improved(n):
	t = time.process_time_ns()

	p_list = []

	it_count = 0

	while (n % 2 == 0):
	    p_list.append(2)
	    n /= 2

	    it_count += 1

	p = Decimal(3)

	while (p*p <= n):
		while (n%p == 0):
			p_list.append(p)
			n /= p

		p += 2
		
		it_count += 1
	
	if (n != 1):
		p_list.append(n)

	if (len(p_list) > 2):
	    print("AAAAAA")
	
	return int(p_list[0]), int(p_list[1]), it_count, time.process_time_ns()-t